LinqSample-master/LinqSamples/JobScheduler/JobScheduler.cs

```diff
@@ -0,0 +1,53 @@
+﻿using AnalyticsAdapter;
+using System;
+using System.Collections.Generic;
+using System.Timers;
+
+namespace JobSchedulers
+{
+    public interface IJob
+    {
+        void Execute(DateTime signalTime);
+    }
+
+    public class JobScheduler : IJob
+    {
+        private readonly Timer _timer;
+        private readonly List<Action<DateTime>> _actions = new();
+
+        public JobScheduler(int intervalMs)
+        {
+            _timer = new Timer(intervalMs);
+            _timer.Elapsed += OnTimedEvent;
+            _timer.AutoReset = true;
+            _timer.Enabled = false;
+        }
+
+        public void AddHandler(Action<DateTime> action)
```

Comment:
и передавать сюда джобу а не экшен
```suggestion
        public void AddHandler(IJob job)
```